File: Roman Numerals Helper/Python/solution_1.py
class RomanNumerals:
    roman_map = ('M', 1000), ('CM', 900), ('D', 500), ('CD', 400), ('C', 100), ('XC', 90), ('L', 50), ('XL', 40), ('X', 10), ('IX', 9), ('V', 5), ('IV', 4), ('I', 1)

    @staticmethod
    def to_roman(num):
        roman = ''
        while num > 0:
            for r, n in RomanNumerals.roman_map:
                while num >= n:
                    roman += r
                    num -= n
        return roman

    @staticmethod
    def from_roman(roman_num):
        num = 0
        for r, n in RomanNumerals.roman_map:
            while roman_num.startswith(r):
                num += n
                roman_num = roman_num[len(r):]
        return num

File: Roman Numerals Helper/JavaScript/solution_1.js
class RomanNumerals {
    static fromRoman(num) {
        const dict = { M: 1000, D: 500, C: 100, L: 50, X: 10, V: 5, I: 1 };
        return num.split('').reduce((acc, curr, i, arr) => {
            const currVal = dict[curr];
            const nextVal = dict[arr[i+1]];
            if (nextVal && currVal < nextVal) return acc - currVal;
            return acc + currVal;
        }, 0);
    }

    static toRoman(num) {
        const roman = [
            ['','I','II','III','IV','V','VI','VII','VIII','IX'], 
            ['','X','XX','XXX','XL','L','LX','LXX','LXXX','XC'], 
            ['','C','CC','CCC','CD','D','DC','DCC','DCCC','CM'], 
            ['','M','MM','MMM']
        ];
        const digits = num.toString().split('').reverse();
        let romanNum = '';
        for (let i = 0; i < digits.length; i++) {
            romanNum = roman[i][parseInt(digits[i])] + romanNum;
        }
        return romanNum;
    }
}
